identity_stardust/src/document/stardust_document.rs

```diff
@@ -311,63 +287,35 @@ impl StardustDocument {
-  pub fn alias_id_to_did(id: &AliasId) -> Result<StardustDID> {
-    // Manually encode to hex to avoid 0x prefix.
-    let hex: String = BaseEncoding::encode(id.as_slice(), Base::Base16Lower);
-    StardustDID::parse(format!("did:stardust:{hex}")).map_err(Into::into)
+    StateMetadataDocument::unpack(state_metadata).and_then(|doc| doc.into_stardust_document(did))
   }
+}
 
-  pub fn did_to_alias_id(did: &StardustDID) -> Result<AliasId> {
-    // TODO: just use 0x in the tag as well?
-    // Prepend 0x manually.
-    AliasId::from_str(&format!("0x{}", did.method_id())).map_err(Into::into)
-  }
+#[cfg(feature = "iota-client")]
+mod stardust_document_iota_client {
+  use iota_client::block::output::Output;
 
-  // TODO: can hopefully remove if the publishing logic is wrapped.
-  pub fn did_from_block(block: &Block) -> Result<StardustDID> {
-    let id: AliasId = AliasId::from(get_alias_output_id_from_payload(block.payload().unwrap()));
-    Self::alias_id_to_did(&id)
-  }
+  use crate::Error;
 
-  /// Deserializes a JSON-encoded `StardustDocument` from an Alias Output block.
-  ///
-  /// NOTE: `did` is required since it is omitted from the serialized DID Document and
-  /// cannot be inferred from the [`Output`]. It also indicates the network, which is not
-  /// encoded in the `AliasId` alone.
-  // TODO: remove? Is `unpack` sufficient?
-  pub fn deserialize_from_output(did: &StardustDID, output: &Output) -> Result<StardustDocument> {
-    let document: &[u8] = match output {
-      Output::Alias(alias_output) => alias_output.state_metadata(),
-      _ => panic!("not an alias output"),
-    };
-    Self::unpack(did, document)
-  }
-}
+  use super::*;
 
-// helper function to get the output id for the first alias output
-// TODO: error handling
-fn get_alias_output_id_from_payload(payload: &Payload) -> OutputId {
-  match payload {
-    Payload::Transaction(tx_payload) => {
-      let TransactionEssence::Regular(regular) = tx_payload.essence();
-      for (index, output) in regular.outputs().iter().enumerate() {
-        if let Output::Alias(_alias_output) = output {
-          return OutputId::new(tx_payload.id(), index.try_into().unwrap()).unwrap();
-        }
-      }
-      panic!("No alias output in transaction essence")
+  impl StardustDocument {
+    /// Deserializes a JSON-encoded `StardustDocument` from an Alias Output block.
+    ///
+    /// NOTE: `did` is required since it is omitted from the serialized DID Document and
+    /// cannot be inferred from the [`Output`]. It also indicates the network, which is not
+    /// encoded in the `AliasId` alone.
+    // TODO: remove? Is `unpack` sufficient?
+    pub fn unpack_from_output(did: &StardustDID, output: &Output) -> Result<StardustDocument> {
+      let document: &[u8] = match output {
+        Output::Alias(alias_output) => alias_output.state_metadata(),
+        _ => return Err(Error::InvalidStateMetadata("not an alias output")),
+      };
+      Self::unpack(did, document)
```

Comment:
It's convenient. I'd leave it in for now and see if we really need it when we write more examples.